bootstrap/bootstrapper/fs/source.go

```diff
@@ -130,6 +120,12 @@ func (fss *fileSystemSource) ReadData(shard uint32, tr xtime.Ranges) (m3db.Shard
 			block.Reset(timeRange.Start, encoder)
 			curMap.AddBlock(id, block)
 		}
+		if !hasError {
+			if err := r.Validate(); err != nil {
+				hasError = true
+				log.Errorf("data validation error: %v", err)
```

Comment:
Perhaps something about the activity to help chase this down if/when someone sees this?  e.g. "error reading data file for shard %d time %v: failed validation, %v", shard, t, err